src/main/java/gmail/taras/shevchuk0024/lesson40_Functional_Interface/Supplier.java

```diff
@@ -0,0 +1,19 @@
+package gmail.taras.shevchuk0024.lesson40_Functional_Interface;
+
+import java.util.ArrayList;
+
+public class Supplier {
+    public static java.util.function.Supplier<String> randomWeekDay = () -> {
+        ArrayList<String> days = new ArrayList<>();
+        days.add("Понеділок");
+        days.add("Вівторок");
+        days.add("Середа");
+        days.add("Четвер");
+        days.add("Пʼятниця");
+        days.add("Субота");
+        days.add("Неділя");
+
+        int value = (int) (Math.random() * days.size());
+        return days.get(value);
+    };
+}
```

Comment:
Тут краще використати статичний масив стрінг фіксованого розміру. Вірогідність що кількість тижнів зміниться **дуже** мала. Коли ми використовуємо ліст то у нас є можливість його розширювати (а це нам не потрібно) тому він і займає більше пам'яті. Плюс кожного разу Ви створюєте один і той сами список із разу в раз коли ви викликаєте лямбду. Краще винести цю змінну як константу щоб джава створила її один раз а потім просто використовувала.

```suggestion
package gmail.taras.shevchuk0024.lesson40_Functional_Interface;

import java.util.ArrayList;
import java.util.function.Supplier;

public class SupplierUtils {

    public static final String[] WEEK_DAYS = {
        "Понеділок",
        "Вівторок",
        "Середа",
        "Четвер",
        "Пʼятниця",
        "Субота",
        "Неділя"
    };

    public static Supplier<String> randomWeekDay = () -> {
        int randomWeekDayIndex = (int) (Math.random() * WEEK_DAYS.length);
        
        return WEEK_DAYS[randomWeekDayIndex];
    };
}
```